W-17588933 sftp max idle (#2771)
mulesoft/docs-connectors: sftp/2.4/modules/ROOT/pages/sftp-connector-upgrade-guide.adoc

If your *File Write* operation overrides instead of appending when using the `APPEND` mode, use the `CUSTOM` write strategy as it isn’t applicable to the `OVERWRITE` or `CREATE_NEW` modes. With this strategy, you can manually specify the offset for the server to write the file in `APPEND` mode, particularly only for older SFTP servers that don’t directly support write modes. Use this strategy only if the `STANDARD` strategy results in errors during file writing in `APPEND` mode and doesn’t offer concurrency. Additionally, you can adjust the buffer size in the *Buffer Size for Write Strategy* field. 

=== Override Max Idle Configuration

> Even when using a custom write strategy, consecutive write calls might experience failures during the `APPEND` write mode. To address this issue, set the *Max Idle* field in the Advanced tab of the SFTP Connection Config to `0`. This change forces the SFTP connection to be recreated for each connection, ensuring that the `APPEND` write mode is successfully executed. However, it is important to note that this workaround incurs a performance impact due to the overhead of frequent connection re-establishment.

Reviewer: ```suggestion
If you use a custom write strategy and experience consecutive write call failures during the `APPEND` write mode, set the *Max Idle* field in the Advanced tab of the SFTP Connection Config to `0`. This change forces the SFTP connection to be recreated for each connection, ensuring that the `APPEND` write mode executes successfully. This workaround incurs a performance impact due to the overhead of frequent connection reestablishment.
```
